### PR TITLE
Convert to NugetExistsChecker to dotnet

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [ ] The **_head(source)_** branch for this pull request is a **_feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: feature/<issue-num>-<description>
      Example: feature/123-my-feature
      ```
    </details>

- [ ] The **_head(source)_** branch for this pull request is created from the **_master_** branch.

- [ ] The **_base(target)_** branch for this pull request is the **_master_**.
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

- [ ] Pull request title matches the title of the linked issue.

- [ ] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [ ] The labels attached to this PR match the labels attached to the associated issue.

- [ ] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out and let's discuss it.

- [ ] I have manually tested my code changes to the best of my ability.

- [ ] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - This does not mean that the tests pass in both the local and CI environment.
  - Status checks run for this pull request when merging into the **_master_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

**Review by KinsonDigitalAdmin — COMMENTED** [review]
Convert all class fields that start with an `_` to use the `this` prefix instead.

File: GotNuget/GitHubAction.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+using GotNuget.Exceptions;
+using GotNuget.Services;
+
+namespace GotNuget;
+
+/// <inheritdoc/>
+public sealed class GitHubAction : IGitHubAction
+{
+    private readonly IGitHubConsoleService _gitHubConsoleService;
```
Comment (KinsonDigitalAdmin) [inline GotNuget/GitHubAction.cs:13]: Refactor all class fields to **NOT** use an `_`. 
Use `this` instead.

Do this across entire code base

File: GotNuget/GitHubAction.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            return;
+        }
+
+        _nugetDataService.Dispose();
+
+        _isDisposed = true;
+    }
+
+    /// <summary>
+    /// Shows a welcome message with additional information.
```
Comment (KinsonDigitalAdmin) [inline GotNuget/GitHubAction.cs:96]: Refactor:

```suggestion
    /// Shows a welcome message.
```

File: GotNuget/Services/IGitHubConsoleService.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    /// <summary>
+    /// Writes the given <paramref name="value"/> to the console as an error GitHub action workflow command.
+    /// </summary>
+    /// <param name="value">The value to write.</param>
+    void WriteError(string value);
+
+    /// <summary>
+    /// Pauses the console.
+    /// </summary>
+    void PauseConsole();
```
Comment (KinsonDigitalAdmin) [inline GotNuget/Services/IGitHubConsoleService.cs:61]: Change this to only exist in debug mode.
Add DEBUG pre-processor directive

File: GotNuget/Services/IDataParserService.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
+﻿// <copyright file="IDataParserService.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace GotNuget.Services;
+
+/// <summary>
+/// Parses XML data.
+/// </summary>
+public interface IDataParserService
```
Comment (KinsonDigitalAdmin) [inline GotNuget/Services/IDataParserService.cs:10]: Remove this interface entirely.  Not used and was accidentally left behind.

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Review by kselena — CHANGES_REQUESTED** [review]

File: Testing/GotNugetTests/Services/ActionOutputServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    #region Method Tests
+    [Fact]
+    public void SetOutputValue_WhenInvoked_SetsOutputValue()
+    {
+        // Arrange
+        var service = CreateService();
+
+        // Act
+        service.SetOutputValue("my-output", "my-value");
```
[the remark this answers is not in the document]
Comment (CalvinWilkinson) [reply Testing/GotNugetTests/Services/ActionOutputServiceTests.cs:30]: Not needed.  Misunderstanding.

File: Testing/GotNugetTests/Services/ActionOutputServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    [Fact]
+    public void SetOutputValue_WithNullOrEmptyOutputName_ThrowsException()
+    {
+        // Arrange
+        var service = CreateService();
+
+        // Act
+        var act = () => service.SetOutputValue(null, It.IsAny<string>());
```
[the remark this answers is not in the document]
Comment (CalvinWilkinson) [reply Testing/GotNugetTests/Services/ActionOutputServiceTests.cs:43]: Not needed.  Misunderstanding.

**Review by kselena — CHANGES_REQUESTED** [review]
Review doc changes that I made.

**Review by KinsonDigitalAdmin — APPROVED** [review]
